gspread/worksheet.py

```diff
@@ -542,40 +542,165 @@ def get_all_records(
 
-        # if no given expected headers, expect all of them
+    def _validate_headers_and_keys_for_get_records_subset(self, keys, expected_headers):
+        """Validates the returned keys and the given expected headers for `get_records_subset`"""
         if expected_headers is None:
             expected_headers = keys
+        else:
+            # validating the given expected headers
+            if not isinstance(expected_headers, list):
+                raise ValueError("expected_headers must be a list")
+            expected_headers_are_unique = len(expected_headers) == len(
+                set(expected_headers)
+            )
+            if not expected_headers_are_unique:
+                raise GSpreadException("the given 'expected_headers' are not uniques")
 
-        # keys must:
-        # - be uniques
-        # - be part of the complete header list
-        # - not contain extra headers
-        expected = set(expected_headers)
-        headers = set(keys)
-
-        # make sure they are uniques
-        if len(expected) != len(expected_headers):
-            raise GSpreadException("the given 'expected_headers' are not uniques")
+        # validating the headers in the worksheet
+        header_row_is_unique = len(keys) == len(set(keys))
+        if not header_row_is_unique:
+            raise GSpreadException("the header row in the worksheet is not unique")
 
-        if not expected & headers == expected:
+        # validating that the expected headers are part of the headers in the worksheet
+        expected_headers_are_in_obtained_headers = all(
+            header in keys for header in expected_headers
+        )
+        if not expected_headers_are_in_obtained_headers:
             raise GSpreadException(
                 "the given 'expected_headers' contains unknown headers: {}".format(
-                    expected - headers
+                    set(expected_headers) - set(keys)
                 )
             )
 
+    def _pad_values_and_keys_for_get_records_subset(self, values, keys, default_blank):
+        """Pads the given values and keys for `get_records_subset` if needed"""
+        values_len = len(values[0])
+        keys_len = len(keys)
+        values_wider_than_keys_by = values_len - keys_len
+        default_blank_in_keys = default_blank in keys
+
+        if ((values_wider_than_keys_by > 0) and default_blank_in_keys) or (
+            values_wider_than_keys_by > 1
+        ):
+            raise GSpreadException(
+                "the header row in the worksheet contains multiple empty cells"
+            )
+        elif values_wider_than_keys_by == 1:
+            keys.append(default_blank)
+        elif values_wider_than_keys_by < 0:
+            values = [row + [""] * (-values_wider_than_keys_by) for row in values]
+
+        return values, keys
+
+    def get_records_subset(
+        self,
+        empty2zero=False,
+        head=1,
+        first_row=None,
+        last_row=None,
```

Comment:
first_row and last_row should be required arguments. Otherwise this is just get_all_records again. Perhaps `head` should also be required

Reply:
I agree, Done.